src/controller/python/chip-device-ctrl.py

```diff
@@ -560,6 +560,74 @@ def do_discover(self, line):
 
         self.devCtrl.PrintDiscoveredDevices()
 
+    def do_ping(self, line):
+        """
+        ping [options] <nodeid>
+
+        Options:
+          -i  <interval>  ping interval time in seconds
+          -c  <count>     stop after <count> replies
+          -r  <1|0>       enable or disable MRP
+          -s  <size>      payload size in bytes
+
+        ping command is using Echo Protocol to measure packet loss across network paths.
+
+        """
+        try:
+            args = shlex.split(line)
+            argc = len(args) - 1
+            if len(args) < 1 or args[0] == '?':
```

Comment:
I have been told in several PR reviews that refactoring command parsing with e.g. argparse is a great idea, but out-of-scope for a given PR.

But here we have introduction of a brand new command, copying the same manual argument parsing. Why are we doing this?  If we're introducing a brand new command, why can't we use a library parser?

Reply:
Can argparse be used on parsing the arguments to those sub-functions? I think it is used on parse the arguments to python program.

Reply:
Let's use argparse ASAP. It also **properly handles printing default values**.

Reply:
This is what I'm suggesting:

```
import argparse
import sys

class ChipTool(object):

    def do_ping(self, line):
        """
        ping [options] <nodeid>
        Options:
          -i  <interval>  ping interval time in seconds
          -c  <count>     stop after <count> replies
          -r  <1|0>       enable or disable MRP
          -s  <size>      payload size in bytes
        ping command is using Echo Protocol to measure packet loss across network paths.
        """
        parser = argparse.ArgumentParser()
        parser.add_argument("-i", type=int, default=1000, dest='interval')
        parser.add_argument("-c", type=int, default=3, dest='count')
        parser.add_argument("-r", type=bool, default=True, dest='mrp')
        parser.add_argument("-s", type=int, default=32, dest='size')
        parser.add_argument('nodeid', type=int, nargs=1)
        args = parser.parse_args(line)
        err = self.devCtrl.PingNode(args['nodeid'], args['count'], args['interval'], args['size'], args['mrp'])

        if err == 0:
            print("Received command status response")
        else:
            print("Failed to receive command status response")

if __name__ == '__main__':

    ChipTool().do_ping(sys.argv[1:])
```

Reply:
> Can argparse be used on parsing the arguments to those sub-functions? I think it is used on parse the arguments to python program.

Yes, as in the above, you can pass an explicit array of arguments to parse_args.  It only defaults to parsing sys.argv if you pass nothing.

Reply:
Acked

Reply:
Thanks for the refactor.  Hopefully others can replicate the pattern when adding new commands.